# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/shared/CodeGenerator-shared.h
# Commit: 09390cf1d667
# Full Hash: 09390cf1d66716146aeb3112f0ac640e317890d9
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-14 09:51:01
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.6: Add tls dependency for truncate i32. r=lth
#   
#   We generate builtin call for MTruncateToInt32 operation for floating points types,
#   so we need to add a tls dependency.
#   I inserted NYI for arm64 because Ion doesn't support arm64.
# ==============================================================================

diff -r 83a3ed07220d -r 09390cf1d667 js/src/jit/shared/CodeGenerator-shared.h
--- a/js/src/jit/shared/CodeGenerator-shared.h	Mon Sep 14 04:00:18 2020 +0000
+++ b/js/src/jit/shared/CodeGenerator-shared.h	Mon Sep 14 05:19:44 2020 +0000
@@ -279,11 +279,16 @@
 
   OutOfLineCode* oolTruncateDouble(
       FloatRegister src, Register dest, MInstruction* mir,
-      wasm::BytecodeOffset callOffset = wasm::BytecodeOffset());
+      wasm::BytecodeOffset callOffset = wasm::BytecodeOffset(),
+      bool preserveTls = false);
   void emitTruncateDouble(FloatRegister src, Register dest,
                           MTruncateToInt32* mir);
+  void emitTruncateDoubleBuiltin(FloatRegister src, Register dest,
+                                 MWasmBuiltinTruncateToInt32* mir);
   void emitTruncateFloat32(FloatRegister src, Register dest,
                            MTruncateToInt32* mir);
+  void emitTruncateFloat32Builtin(FloatRegister src, Register dest,
+                                  MWasmBuiltinTruncateToInt32* mir);
 
   void emitPreBarrier(Register elements, const LAllocation* index);
   void emitPreBarrier(Address address);